### PR TITLE
DROOLS-2978 : Test Scenario: unable to select rule defined in xlsx in EXPECTATION settings

### PR DESCRIPTION
**Pull request by Rikkola**
https://issues.jboss.org/browse/DROOLS-2978

### CODE REVIEW

**Comment by Rikkola** [comment]
@manstis Good afternoon, you might be the best to take a look.

File: drools-wb-screens/drools-wb-dtable-xls-editor/drools-wb-dtable-xls-editor-backend/src/test/java/org/drools/workbench/screens/dtablexls/backend/server/indexing/IndexDecisionTableXLSXTest.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+import org.kie.workbench.common.services.refactoring.service.ResourceType;
+import org.uberfire.ext.metadata.io.KObjectUtil;
+import org.uberfire.java.nio.file.Path;
+
+public class IndexDecisionTableXLSXTest extends BaseIndexingTest<DecisionTableXLSXResourceTypeDefinition> {
+
+    @Test
+    public void testIndexDecisionTableXLSMultipleTypes() throws IOException, InterruptedException {
+        //Add test files
+        final Path path3 = loadXLSFile(basePath,
```
Comment (manstis) [inline drools-wb-screens/drools-wb-dtable-xls-editor/drools-wb-dtable-xls-editor-backend/src/test/java/org/drools/workbench/screens/dtablexls/backend/server/indexing/IndexDecisionTableXLSXTest.java]: Minor comment; any reason you want to call the _single_ ```Path``` object ```path3```?

Comment (Rikkola) [reply drools-wb-screens/drools-wb-dtable-xls-editor/drools-wb-dtable-xls-editor-backend/src/test/java/org/drools/workbench/screens/dtablexls/backend/server/indexing/IndexDecisionTableXLSXTest.java]: Eeeeeeeemmm, can't think of a good reason.

**Comment by jomarko** [comment]
Please, do not track `jacoco.exec` file.

File: drools-wb-screens/drools-wb-dtable-xls-editor/drools-wb-dtable-xls-editor-backend/src/main/java/org/drools/workbench/screens/dtablexls/backend/server/indexing/DecisionTableXLSXFileIndexer.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            return fillDrlIndexBuilder(path, drl);
+        } finally {
+            if (inputStream != null) {
+                try {
+                    inputStream.close();
+                } catch (IOException e) {
+                    // no-op
+                }
+            }
+        }
```
Comment (jomarko) [inline drools-wb-screens/drools-wb-dtable-xls-editor/drools-wb-dtable-xls-editor-backend/src/main/java/org/drools/workbench/screens/dtablexls/backend/server/indexing/DecisionTableXLSXFileIndexer.java]: Worth to use try with resources:

```java
try (final InputStream inputStream = ioService.newInputStream(path, StandardOpenOption.READ)) {
            final String drl = DecisionTableFactory.loadFromInputStream(inputStream,
                                                                        null);

            return fillDrlIndexBuilder(path, drl);
        }
```

File: drools-wb-screens/drools-wb-dtable-xls-editor/drools-wb-dtable-xls-editor-backend/src/test/java/org/drools/workbench/screens/dtablexls/backend/server/indexing/IndexDecisionTableXLSXTest.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+        {
+            final Query query = new SingleTermQueryBuilder(
+                    new ValueResourceIndexTerm("*", ResourceType.RULE, ValueIndexTerm.TermSearchType.WILDCARD)
+
+            )
+                    .build();
+            // Rule name found from xlsxrule
+            searchFor(index, query, 1, path);
+        }
```
Comment (jomarko) [inline drools-wb-screens/drools-wb-dtable-xls-editor/drools-wb-dtable-xls-editor-backend/src/test/java/org/drools/workbench/screens/dtablexls/backend/server/indexing/IndexDecisionTableXLSXTest.java]: Could you please explain why we need this separate block?

**Comment by Rikkola** [comment]
@manstis Hi, are you happy with the changes or would you still like to see some changes?

**Comment by manstis** [comment]
@Rikkola Nope, just needed a nudge.

**Pull request merged**